### PR TITLE
fix(list/chart views): Chart Properties modal now has transitions

### LINKED ISSUE #11520: Inconsistent Behavior for CRUD view modals
## Screenshot

dashboard-modal

## Description

In the new React-based Dashboard & Chart CRUD view. The edit modal has animation on both opening and closing, but the edit modal has only opening animation.

Also, I think we should probably consider linking the edit button to opening `<dashboard_url?>edit=true` in a new window instead of opening the properties modal, because it's much more commonly needed to edit the dashboard layout/charts rather than just the things in properties modal.

## Design input
[describe any input/collaboration you'd like from designers, and
tag accordingly. For design review, add the
label `design:review`. If this includes a design proposal,
include the label `design:suggest`]

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY
Fixes: https://github.com/apache/superset/issues/11520

The properties modal was previously conditionally rendered, rather than using the visibility prop. This fixes it, and adds a bunch of safe navigation operators to handle the default (null) slice scenario.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/features/home/ChartTable.tsx [hunk 1]
```diff
@@ -171,15 +171,12 @@ function ChartTable({
...
-          onSave={handleChartUpdated}
-          show
-          slice={sliceCurrentlyEditing}
-        />
-      )}
-
+      <PropertiesModal
+        onHide={closeChartEditModal}
+        onSave={handleChartUpdated}
+        show
```
Comment (justinpark) [inline superset-frontend/src/features/home/ChartTable.tsx:177]: it appears missing the `show` condition
```suggestion
        show={sliceCurrentlyEditing}
```

**Review by mistercrunch — APPROVED** [review]
LGTM

**Comment by sadpandajoe** [comment]
@supersetbot label 4.1

**Comment by sadpandajoe** [comment]
@supersetbot unlabel 4.1

**Comment by sadpandajoe** [comment]
@rusackas looks like when I have this PR in my branch, whenever I go to the chart list I see a superset error that needs to be be dismissed.

**Pull request merged**